src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/model/SentenceCalculation.kt

```diff
@@ -266,29 +310,90 @@ data class SentenceCalculation(
       if (sentence.calculateErsedFromTwoThirds()) {
         return calculateErsedFromTwoThirds()
       }
+      if (sentence.calculateMixed()) {
+        return calculateErsedMixed()
+      }
     }
     return null
   }
 
+  private fun calculateErsedMixed(): ReleaseDateCalculationBreakdown? {
+    val days = if (sentence is ConsecutiveSentence) {
+      ConsecutiveSentenceAggregator((sentence as ConsecutiveSentence).orderedSentences.map { it.custodialDuration() }).calculateDays(
+        sentence.sentencedAt,
+      )
+    } else {
+      val custodialDuration = sentence.custodialDuration()
+      custodialDuration.getLengthInDays(sentence.sentencedAt)
+    }
+
+    val effectiveRelease = extendedDeterminateParoleEligibilityDate ?: adjustedDeterminateReleaseDate
```

Comment:
It looks like this is duplicated in a few places. Where the ersed is calculated by the max period before release. Might be good to reduce the duplication in a new function.

```
val release = extendedDeterminateParoleEligibilityDate ?: releaseDate
      val ersed = release.minusDays(MAX_ERSED_PERIOD_DAYS.toLong())
      ReleaseDateCalculationBreakdown(
        rules = setOf(CalculationRule.ERSED_MAX_PERIOD),
        releaseDate = ersed,
        unadjustedDate = unadjustedExtendedDeterminateParoleEligibilityDate ?: unadjustedDeterminateReleaseDate,
        adjustedDays = ChronoUnit.DAYS.between(
          unadjustedDeterminateReleaseDate,
          adjustedDeterminateReleaseDate,
        ).toInt(),
        rulesWithExtraAdjustments = mapOf(
          CalculationRule.ERSED_MAX_PERIOD to AdjustmentDuration(
            -MAX_ERSED_PERIOD_DAYS,
            ChronoUnit.DAYS,
          ),
        ),
      )
      ```

Reply:
Ordinarily - we would refactor this. I think its more important we get this into dev at the moment given the timeline.